### DISCUSSION TITLE
32 bit hardware - Pentium 4 etc

### DISCUSSION THREAD

**Discussion by PartialVolume** [PartialVolume/shredos.x86_64#73] [Show and tell]
I'm building a 32 bit .img & .iso of the latest release v2021.08.2_20_0.32.014, just wondered if you or anybody else has any 32 bit systems that have a working USB & CDROM to test the 32 bit build?

No problems, if not, it's just that I won't be able to get to my old Pentium test systems for about a week or so.

@Firminator et.al

**Comment by Firminator** [comment]
I might be able to test it on an old PowerEdge 2500 w/ Pentium III !?!? ALso have an old OptiPlex GX620 (not sure what CPU it has).

Wikipedia surely helps to find out which CPU is 32bit:
>The first processor to implement Intel 64 was the multi-socket processor Xeon code-named Nocona in June 2004. In contrast, the initial Prescott chips (February 2004) did not enable this feature. Intel subsequently began selling Intel 64-enabled Pentium 4s using the E0 revision of the Prescott core, being sold on the OEM market as the Pentium 4, model F. The E0 revision also adds eXecute Disable (XD) (Intel's name for the NX bit) to Intel 64, and has been included in then current Xeon code-named Irwindale. Intel's official launch of Intel 64 (under the name EM64T at that time) in mainstream desktop processors was the N0 stepping Prescott-2M.
>
>The first Intel mobile processor implementing Intel 64 is the Merom version of the Core 2 processor, which was released on July 27, 2006. **None of Intel's earlier notebook CPUs (Core Duo, Pentium M, Celeron M, Mobile Pentium 4) implement Intel 64.**
>
>Implementations
Intel's processors implementing the Intel64 architecture include the Pentium 4 F-series/5x1 series, 506, and 516, Celeron D models 3x1, 3x6, 355, 347, 352, 360, and 365 and all later Celerons, all models of Xeon since "Nocona", all models of Pentium Dual-Core processors since "Merom-2M", the Atom 230, 330, D410, D425, D510, D525, N450, N455, N470, N475, N550, N570, N2600 and N2800, all versions of the Pentium D, Pentium Extreme Edition, Core 2, Core i9, Core i7, Core i5, and Core i3 processors, and the Xeon Phi 7200 series processors.

I'll check the CPU of the GX620. I believe we phased those out years ago due to that exact reason: they were 32bit, but I'll check to make sure.

**Comment by PartialVolume** [comment]
That's great, thanks,  I have an early P4 32bit and a P3, that I'll be able to try next week.

One of these days I'll try building a ShredOS SPARC 64 version. I've got a couple of Sun Blades 150 and Sun Ultra 10. It would be cool getting it to boot via USB on a Blade 150.

Amazingly I still have one or two 486 systems running DOS that are still in production use, working away on site. ( Don't ask :-) ShredOS is way too large for a 486 though.

The 32 bit build should be ready tomorrow, I'll send you a download link when it's done.

**Reply by Firminator** [reply]
>486

LOL that would be a wonderful proof-of-concept. Take out all the UEFI bloat, remove a bunch of video drivers, remove RAID controller drivers. Keep it to the very mimimum.

>Sun Blades 150 and Sun Ultra 10

I don't know these systems, but if they don't support booting from USB then your ISO would come in handy. If they have USB ports, but the BIOS does not support booting from USB then you could use PLOP[**1**] as a workaround. I used this in the past to enable the PowerEdge 2500 to boot from USB :)


[**1**] PLOP boots a physical CD/ISO from CDROM and then chainloads/emulates booting from USB. See https://www.plop.at/en/bootmanager/index.html

**Comment by PartialVolume** [comment]
That plop bootloader looks interesting. But before I find I'm getting myself side tracked, I need to stay focused in getting ShredOS/nwipes next big features implemented for the next release. HPA/DSO and secure erase.

I think HPA/DSO and secure erase will be an important update.

I just have one or two other things I have to do first.
- Release the 32bit img & .iso
- Release nwipe 0.33
- Document the build procedure.

**Comment by PartialVolume** [comment]
@Firminator Slight change of plan, while testing the 32bit build on a 64bit machine, I discovered a couple of minor bugs in nwipe that are only a problem in a 32 bit build. I'll create a couple of issues on nwipe then work on those next week when I have a proper 32 bit machine to compile & test on. 

So the 32bit build .img/.iso testing will be delayed for a week or so. I'll also delay the release of nwipe 0.33 and roll the two fixes into that. In the meantime I'll start on the documentation instead.

**Reply by Firminator** [reply]
Sounds good to me. Excited about the documentation about ho to compile ShredOS with the correct settings.
